rasa/core/actions/forms.py

```diff
@@ -370,22 +389,50 @@ async def request_next_slot(
         domain: Domain,
         output_channel: OutputChannel,
         nlg: NaturalLanguageGenerator,
+        events_so_far: List[Event],
     ) -> List[Event]:
         """Request the next slot and utter template if needed,
             else return None"""
 
-        for slot in self.required_slots(domain):
-            if self._should_request_slot(tracker, slot):
-                logger.debug(f"Request next slot '{slot}'")
+        request_slot_events = []
+
+        # If this is not `None` it means that the custom action specified a next slot
+        # to request
+        slot_to_request = next(
+            (
+                event.value
+                for event in events_so_far
+                if isinstance(event, SlotSet) and event.key == REQUESTED_SLOT
+            ),
+            None,
+        )
+        temp_tracker = self._temporary_tracker(tracker, events_so_far, domain)
 
-                bot_message_events = await self._ask_for_slot(
-                    domain, nlg, output_channel, slot, tracker
-                )
-                return [SlotSet(REQUESTED_SLOT, slot), *bot_message_events]
+        if not slot_to_request:
+            slot_to_request = self._find_next_slot_to_request(temp_tracker, domain)
+            request_slot_events.append(SlotSet(REQUESTED_SLOT, slot_to_request))
+
+        if slot_to_request:
+            bot_message_events = await self._ask_for_slot(
+                domain, nlg, output_channel, slot_to_request, temp_tracker
+            )
+            return request_slot_events + bot_message_events
 
         # no more required slots to fill
         return [SlotSet(REQUESTED_SLOT, None)]
 
+    def _find_next_slot_to_request(
+        self, tracker: DialogueStateTracker, domain: Domain
+    ) -> Optional[Text]:
+        return next(
+            (
+                slot
+                for slot in self.required_slots(domain)
+                if self._should_request_slot(tracker, slot)
+            ),
+            None,
+        )
+
     def _name_of_utterance(self, domain: Domain, slot_name: Text) -> Text:
         full_name = f"utter_ask_{self._form_name}_{slot_name}"
```

Comment:
if the name starts with `utter_` it is not a custom action, should we first look for `full_name = f"action_ask_{self._form_name}_{slot_name}"` and only `run` in this case?

Reply:
Let's do this after the alpha and see if users actually want a custom action to ask for the next slot.

Reply:
ok, let's remove `.run` there and add `TODO`

Reply:
The run is fine, what speaks against it? It will use `ActionUtterTemplate` and that's it?

Reply:
it just a bit counter intuitive (form how usually we call templates in custom actions), but I think it is actually fine